src/Avatar/Avatar.tsx

```diff
@@ -31,18 +28,31 @@ const StyledAvatar = styled.img.attrs<StyledAvatarProps>(props => ({
   overflow: hidden; // Ensure page layout in Firefox should images fail to load
   line-height: ${get('lineHeights.condensedUltra')};
   vertical-align: middle;
-  border-radius: ${props => getBorderRadius(props)};
+  // If the avatar is square and size is greater than 24px (at any breakpoint), border-radius will be 6px. Otherwise, it will be 4px.
+  border-radius: ${props => (props.square ? 'clamp(4px, var(--avatar-size) - 24px, 6px)' : '50%')};
   box-shadow: 0 0 0 1px ${get('colors.avatar.border')};
+  height: var(--avatar-size);
+  width: var(--avatar-size);
   ${sx}
 `
 
 export type AvatarProps = ComponentProps<typeof StyledAvatar>
 
 const Avatar = React.forwardRef<HTMLImageElement, AvatarProps>(function Avatar(
-  {alt = '', size = 20, square = false, ...rest},
+  {alt = '', size = DEFAULT_AVATAR_SIZE, square = false, sx: sxProp = defaultSxProp, ...rest},
   ref,
 ) {
-  return <StyledAvatar ref={ref} alt={alt} size={size} square={square} {...rest} />
+  const avatarSx = isResponsiveValue(size)
+    ? merge(
```

Comment:
Sorry I missed your ping about the type stuff.

I think we could also do 

```
merge<BetterSystemStyleObject>(...)
```

instead of typing each input `as`?

Also totally optional maybe for readability 

```jsx
const avatarSx = isResponsiveValue(size)
    ? getBreakpointDeclarations(
        size,
        '--avatar-size' as keyof React.CSSProperties,
        value => `${value || DEFAULT_AVATAR_SIZE}px`,
      )
    : {'--avatar-size': `${size}px`}
```

and then in the `StyledAvatar`

```
sx={merge<BetterSystemStyleObject>(avatarSx, sxProp)}
```

Just a suggestion - however you prefer.

Reply:
I _think_ I remember there being an performance issue with passing `merge()` directly to to `sx`, so I'm going to keep it as-is. Definitely going to use a generic instead of `as` though. Great suggestion!

Reply:
Ohh I didn't know about the performance impact on using merge in sx! Thanks for letting me know!